edx_repo_tools/codemods/django3/github_actions_modernizer_django.py

```diff
@@ -0,0 +1,56 @@
+"""
+Django Matrix Modernizer for Github Actions CI
+"""
+import re
+from copy import deepcopy
+
+import click
+
+from edx_repo_tools.utils import YamlLoader
+
+DJANGO_ENV_PATTERN = r"django[0-3][0-2][0-2]?"
+DJANGO_VER_PATTERN = re.compile(r"[2-3]\.[0-3]")
+DJANGO_VERSIONS = ['2.2', '3.0', '3.1', '3.2']
+ALLOWED_DJANGO_ENVS = ['django22', 'django30', 'django31', 'django32']
+
+
+class GithubCIDjangoModernizer(YamlLoader):
+    def __init__(self, file_path):
+        super().__init__(file_path)
+        self.yml_instance.default_flow_style = None
+        self.yml_instance.indent(mapping=2, sequence=2, offset=0)
+
+    def _update_matrix(self):
+        for section_key in self.elements['jobs'].keys():
+            matrix_items = self.elements['jobs'][section_key]['strategy']['matrix']
+
+            for key, values in matrix_items.items():
+                if 'django' in key and any(DJANGO_VER_PATTERN.match(value) for value in values):
+                    self.elements['jobs'][section_key]['strategy']['matrix'][key] = deepcopy(DJANGO_VERSIONS)
+
+                if 'tox' in key:
+                    matrix_items = self.elements['jobs'][section_key]['strategy']['matrix'][key]
+                    has_django_env = any(re.search(DJANGO_ENV_PATTERN, item) for item in matrix_items)
```

Comment:
with this value based filter in place, do we need `if 'django' in key` and `if 'tox' in key`?

Reply:
since we're already extracting non-django environments and appending them with updated django environments

Reply:
We need these checks to prevent the `Python` versions to be upgraded since they also match the regex of Django versions.

Reply:
why would a `python` version match the Django pattern?

Reply:
I was considering the future scenario that if we need to add `Python 4.0` and `Django 4.0` version support, then we won't have to modify the functionality and will only need to update the regex for the required version. 
Adding `django in key` check also prevents the failure in case we have the `include` and `exclude` options added in the matrix. The current functionality will automatically skip those cases.

Reply:
the pattern should have `django` in start and eliminate the possibility of matching it with any python environment

Reply:
and we don't have any use case where we've specified `2.2` instead of `django22`

Reply:
We have two patterns. One for matching the `DJANGO_VERSIONS` i.e. `[2.2, 3.0, 3.1, 3.2]` and one for matching `DJANGO_ENVS` i.e. `[django22, django30, django31, django32]` in the toxenvs. 
The current check is to handle the change in case of the first pattern.

Reply:
there's no need for `DJANGO_VERSIONS` pattern

Reply:
We need the `DJANGO_VER_PATTERN` to handle the case of `django-versions` option in the matrix and `DJANGO_ENV_PATTERN` to handle the `toxenvs` in the matrix. 
Both of these are being used in CI files in different repos so we need to handle both of these cases.

Reply:
Since `DJANGO_VER_PATTERN` wasn't needed in our case so removed it and updated the script and tests accordingly.